### PR TITLE
Raises a warning for provide_context instead of killing the task

### PR DESCRIPTION
**Pull request by dimberman**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by ashb — DISMISSED** [review]
This should be a Depreciation warning, and set `stacklevel` so thw warning is issued from the caller's line number, not in here

File: airflow/operators/python.py [hunk 1]
```diff
@@ -85,6 +86,9 @@ def __init__(
         templates_exts: Optional[List[str]] = None,
         **kwargs
     ) -> None:
+        if kwargs.get("provide_context"):
+            warnings.warn("provide_context is deprecated as of 2.0 and is no longer required")
+            kwargs.pop('provide_context', None)
```
Comment (kaxil) [inline airflow/operators/python.py]:
```suggestion
        if kwargs.get("provide_context"):
            warnings.warn("provide_context is deprecated as of 2.0 and is no longer required", DeprecationWarning, stacklevel=2)
            kwargs.pop('provide_context', None)
```

This should take care of @ashb 's suggestion

**Comment by kaxil** [comment]
Just an FYI for future reviewers, we already have the following line in `Updating.md`:

> The change is backwards compatible, setting provide_context will add the provide_context variable to the kwargs (but won't do anything).

https://github.com/apache/airflow/blob/master/UPDATING.md#airflowoperatorspythonpythonoperator

**Review by mik-laj — APPROVED** [review]
Mypy and Jetbrains IDEs don't support deprecation warning for parameters.

**Pull request merged**